tsfc/loopy.py

```diff
@@ -221,6 +254,9 @@ def generate(impero_c, args, scalar_type, kernel_name="loopy_kernel", index_name
                            seq_dependencies=True, silenced_warnings=["summing_if_branches_ops"],
                            lang_version=(2018, 2))
 
+    if PETSc.Log.getActive():
+        knl = lp.register_preamble_generators(knl, [_PreambleGen("#include <petsclog.h>")])
```

Comment:
Depending on if we decide that the faffing with declaring the events outside the C kernels in python and setting the event ids from the outside in the DLLs is worth it, we want to declare the events here in the preamble.